### PR TITLE
add Tables.jl interface

### PR DESCRIPTION
**Pull request by rafaqz**
This PR adds a standard Tables.jl interface and removes Requires.jl and DataFrames code, as it's no longer needed. 

Now a CSV can be saved directly from  a `GBIFRecords` object without DataFrames.jl being involved.

A `DataFrame` will now include all fields as-is form the GBIFRecord (which may be up for debate, but e.g. `basisOfRecord` can be good to have in a table...), as well as the taxon fields previously included. 

This is done by adding methods to `propertynames` and `getproperty`, which means a `GBIFRecord` also has the same properties accessible with the `record.property` syntax. To get the actual taxon object, you use `getfield(record, :taxon)`. Or we can add a wrapper method.

### CODE REVIEW

**Comment by rafaqz** [comment]
The error seems to be an issue with DataFrames and CategoricalArrays? wierd

**Comment by tpoisot** [comment]
Oh it's what is making the tests fail in #42 - I like the idea (@gabrieldansereau might like it even more as he's our resident "export thing as CSV" evangelist).

If the issue really is 1.5, we might make this a major version and set 1.6 as the lower bound.

**Comment by rafaqz** [comment]
The docs on all these PRs also have a problem with QT not finding the `xcb` library to run visually without X on linux.

**Comment by rafaqz** [comment]
The problem was just old dependencies in the tests. I'm not sure why Pkg chose them, but adding fixed versions like 1 for DataFrames seems to have fixed the issue. This is good to go.

**Comment by rafaqz** [comment]
I would really like to merge this... but it needs review some. 

Specifically it may break everything that built off dataframes output before because it includes most of the fields now, and some appear to have previously been named differently to GBIF, and to the struct field names? Now there is no distinction.

**Comment by tpoisot** [comment]
Can you rebase on the main branch? The type piracy issue is gone

**Comment by tpoisot** [comment]
I really want a review from @gabrieldansereau on this - if it's going to break things in your workflows, then maybe we shouldn't merge it yet.

**Comment by tpoisot** [comment]
I also won't merge unless properly documented - I'll start my review when all the elements are here.

**Review by gabrieldansereau — APPROVED** [review]
All good for me. It did not break anything with the few scripts I tried

**Comment by rafaqz** [comment]
> I also won't merge unless properly documented - I'll start my review when all the elements are here.

Before I start, can I clarify what you want documented? Nothing in the current docs appears to be changed by this PR. 

We could however add a listing of available table columns, which are not currently documented.

**Comment by tpoisot** [comment]
> We could however add a listing of available table columns, which are not currently documented.

Yep

File: src/types/GBIFRecords.jl
Comment (tpoisot) [inline src/types/GBIFRecords.jl:44]: Shouldn't this be an `enum`?

File: docs/make.jl
Comment (tpoisot) [inline docs/make.jl]: This should go in the workflow file

**Review by tpoisot — CHANGES_REQUESTED** [review]

File: Project.toml
Comment (tpoisot) [inline Project.toml:10]: Please update the package version number as well

**Review by tpoisot — CHANGES_REQUESTED** [review]

File: src/types/GBIFRecords.jl
Comment (rafaqz) [reply src/types/GBIFRecords.jl:44]: Sorry I don't understand why. It's just used in `propertynames` and `getproperty` as a tuple of `Symbol` that's known at compile time. Because `getproperty` is modified to include the properties that are in `taxon`, as the Tables.jl interface uses `getpropery`.

I would actually prefer not to have taxon at all and remove this complexity completely, generating `GBIFTaxon` when needed instead. But I wasn't sure of the reasoning behind it. Could `taxon` be generated on the fly from `GBIFRecords`?

Comment (tpoisot) [reply src/types/GBIFRecords.jl:44]: It might be generated on the fly _but_ we want to have a `taxon` object, and be able to use the species API. I'm not sure I get your point.

Comment (rafaqz) [reply src/types/GBIFRecords.jl:44]: What I mean is does `taxon` have to be an object actually in the memory layout of `GBIFRecord` like it is curently? 

Instead we could define a constructor and do `GBIFTaxon(record)` whenever we need it for the species API, as its essentially the same thing. Then a lot of this code could be deleted.

Comment (rafaqz) [reply src/types/GBIFRecords.jl:44]: Any interest in this change? it will be a lot cleaner. 

I can write it up fairly easily.

**Comment by rafaqz** [comment]
This is documented and ready to go, besides possible changes to `GBIFRecord` that would make the code a lot simpler.

**Pull request closed without merging**
